File: miniprogram/src/types/api/transfer.ts
/**
 * 调宿申请类型定义
 * @module types/api/transfer
 */

import type { IBaseEntity, IPageParams, IPageResult } from './common';

/**
 * 调宿申请信息
 */
export interface ITransfer extends IBaseEntity {
  /** 学生ID */
  studentId: number;
  /** 学生姓名（冗余） */
  studentName?: string;
  /** 学号（冗余） */
  studentNo?: string;
  /** 原校区编码 */
  originalCampusCode?: string;
  /** 原楼层编码 */
  originalFloorCode?: string;
  /** 原房间ID */
  originalRoomId?: number;
  /** 原房间编码 */
  originalRoomCode?: string;
  /** 原床位ID */
  originalBedId?: number;
  /** 原床位编码 */
  originalBedCode?: string;
  /** 目标校区编码 */
  targetCampusCode?: string;
  /** 目标楼层编码 */
  targetFloorCode?: string;
  /** 目标房间ID */
  targetRoomId?: number;
  /** 目标房间编码 */
  targetRoomCode?: string;
  /** 目标床位ID */
  targetBedId?: number;
  /** 目标床位编码 */
  targetBedCode?: string;
  /** 申请日期 */
  applyDate: string;
  /** 调宿日期 */
  transferDate?: string;
  /** 状态 */
  status: number;
  /** 审核人ID */
  approverId?: number;
  /** 审核人姓名 */
  approverName?: string;
  /** 审核时间 */
  approveTime?: string;
  /** 审核意见 */
  approveOpinion?: string;
  /** 调宿原因 */
  transferReason?: string;
  /** 备注 */
  remark?: string;
}

/**
 * 调宿申请查询参数
 */
export interface ITransferQueryParams extends IPageParams {
  /** 学生ID */
  studentId?: number;
  /** 学号 */
  studentNo?: string;
  /** 学生姓名 */
  studentName?: string;
  /** 原楼层编码 */
  originalFloorCode?: string;
  /** 目标楼层编码 */
  targetFloorCode?: string;
  /** 状态 */
  status?: number;
  /** 申请日期开始 */
  applyDateStart?: string;
  /** 申请日期结束 */
  applyDateEnd?: string;
}

/**
 * 调宿申请提交参数
 */
export interface ITransferSubmitParams {
  /** 目标校区编码 */
  targetCampusCode?: string;
  /** 目标楼层编码 */
  targetFloorCode?: string;
  /** 目标房间ID */
  targetRoomId?: number;
  /** 目标床位ID */
  targetBedId?: number;
  /** 调宿日期 */
  transferDate: string;
  /** 调宿原因 */
  transferReason: string;
}

/**
 * 调宿申请审批参数
 */
export interface ITransferApprovalParams {
  /** 申请ID */
  id: number;
  /** 审批结果 2=通过 3=拒绝 */
  status: 2 | 3;
  /** 分配的目标校区编码 */
  targetCampusCode?: string;
  /** 分配的目标楼层编码 */
  targetFloorCode?: string;
  /** 分配的目标房间ID */
  targetRoomId?: number;
  /** 分配的目标床位ID */
  targetBedId?: number;
  /** 审批意见 */
  approveOpinion?: string;
}

/**
 * 调宿申请分页响应
 */
export type ITransferPageResult = IPageResult<ITransfer>;

File: Application/src/main/java/com/project/backend/accommodation/entity/Transfer.java
package com.project.backend.accommodation.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.project.core.entity.BaseEntity;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 调宿管理实体
 * 
 * @author 陈鸿昇
 * @since 2026-01-06
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("sys_transfer")
@Schema(description = "调宿管理实体")
public class Transfer extends BaseEntity {

    @Schema(description = "学生ID")
    @TableField("student_id")
    private Long studentId;

    @Schema(description = "学生姓名（冗余）")
    @TableField("student_name")
    private String studentName;

    @Schema(description = "学号（冗余）")
    @TableField("student_no")
    private String studentNo;

    @Schema(description = "原校区编码")
    @TableField("original_campus_code")
    private String originalCampusCode;

    @Schema(description = "原楼层编码")
    @TableField("original_floor_code")
    private String originalFloorCode;

    @Schema(description = "原房间ID")
    @TableField("original_room_id")
    private Long originalRoomId;

    @Schema(description = "原房间编码")
    @TableField("original_room_code")
    private String originalRoomCode;

    @Schema(description = "原床位ID")
    @TableField("original_bed_id")
    private Long originalBedId;

    @Schema(description = "原床位编码")
    @TableField("original_bed_code")
    private String originalBedCode;

    @Schema(description = "目标校区编码")
    @TableField("target_campus_code")
    private String targetCampusCode;

    @Schema(description = "目标楼层编码")
    @TableField("target_floor_code")
    private String targetFloorCode;

    @Schema(description = "目标房间ID")
    @TableField("target_room_id")
    private Long targetRoomId;

    @Schema(description = "目标房间编码")
    @TableField("target_room_code")
    private String targetRoomCode;

    @Schema(description = "目标床位ID")
    @TableField("target_bed_id")
    private Long targetBedId;

    @Schema(description = "目标床位编码")
    @TableField("target_bed_code")
    private String targetBedCode;

    @Schema(description = "申请日期")
    @TableField("apply_date")
    private LocalDate applyDate;

    @Schema(description = "调宿日期")
    @TableField("transfer_date")
    private LocalDate transferDate;

    @Schema(description = "状态：1待审核 2已通过 3已拒绝 4已完成")
    @TableField("status")
    private Integer status;

    @Schema(description = "审批实例ID")
    @TableField("approval_instance_id")
    private Long approvalInstanceId;

    @Schema(description = "审核人ID")
    @TableField("approver_id")
    private Long approverId;

    @Schema(description = "审核人姓名")
    @TableField("approver_name")
    private String approverName;

    @Schema(description = "审核时间")
    @TableField("approve_time")
    private LocalDateTime approveTime;

    @Schema(description = "审核意见")
    @TableField("approve_opinion")
    private String approveOpinion;

    @Schema(description = "调宿原因")
    @TableField("transfer_reason")
    private String transferReason;

    @Schema(description = "备注")
    @TableField("remark")
    private String remark;
}
